emukit/quadrature/acquisitions/uncertainty_sampling.py

```diff
@@ -11,63 +11,71 @@
+        :return: Values of the acquisition function at x and unweighted variances. Both shape (num_points x 1).
         """
         variances = self.model.predict(x)[1]
         if self.model.measure is None:
             return variances, variances
         else:
             weights = self.model.measure.compute_density(x).reshape(variances.shape)
-            return variances * weights, variances
+            return variances * weights ** self._measure_power, variances
 
     def evaluate(self, x: np.ndarray) -> np.ndarray:
-        """
-        Evaluates the acquisition function. If a probability measure is used, then the variance will be weighted by
-        the density of the probability measure.
+        """Evaluate the acquisition function.
 
-        :param x: (n_points x n_dims) array of points at which to calculate acquisition function values
-        :return: (n_points x 1) array of acquisition function values
+        :param x: Locations at which to evaluate the acquisition function, shape (num_points x num_dim).
+        :return: Values of the acquisition function at x, shape (num_points x 1).
         """
         return self._evaluate(x)[0]
 
     def evaluate_with_gradients(self, x: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
-        """
-        Optional abstract method that must be implemented if has_gradients returns True.
-        Evaluates value and gradient of acquisition function at x.
+        """Evaluate the acquisition function and compute its gradients.
 
-        :param x: (n_points x n_dims) array of points at which to calculate acquisition function values and gradient
-        :return: Tuple contains an (n_points x 1) array of acquisition function values and (n_points x n_dims) array of
-                 acquisition function gradients with respect to x
+        :param x: Locations at which to evaluate the acquisition function, shape (num_points x num_dim).
+        :return: Values of the acquisition function at x, shape (num_points x 1),
+                 and corresponding gradients, shape (num_points, num_dim).
         """
+        p = self._measure_power
         variance_weighted, variance = self._evaluate(x)
 
         variance_gradient = self.model.get_prediction_gradients(x)[1]
         if self.model.measure is None:
             return variance, variance_gradient
-        else:
-            density = self.model.measure.compute_density(x)
-            density_gradient = self.model.measure.compute_density_gradient(x)
+
+        density = self.model.measure.compute_density(x)
+        density_gradient = self.model.measure.compute_density_gradient(x)
+
+        if p == 1:
             gradient_weighted = (density * variance_gradient.T).T + (variance[:, 0] * density_gradient.T).T
             return variance_weighted, gradient_weighted
+
+        gradient_weighted = (density ** p * variance_gradient.T
+                             + p * (variance[:, 0] * density ** (p - 1)) * density_gradient.T).T
```

Comment:
I think that would benefit from other formatting, or braces, or anything breaking it into smaller chunks. It's hard for me to immediately figure out the order of evaluation of all these operators.

Reply:
I put braces around each term in the sum now. Hope this improves readability.